minio/mcs
# Add Create Service Account api

adds new functionality for creating a service
account for a user, for this, an admin client
is created with the user credentials so that
the service account can be assigned to him.

To test this:
- create a user and assign a policy to him
- login to mcs with the user credentials
- create a serviceAccount on `/api/v1/service-accounts`
  The payload can (optional) have a policy defined as string e.g. 
  ```
   {
	"policy": "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Allow\",\"Action\":[\"s3:GetBucketLocation\"],\"Resource\":[\"arn:aws:s3:::*\"]},{\"Effect\":\"Allow\",\"Action\":[\"s3:ListBucket\",\"s3:GetObject\"],\"Resource\":[\"arn:aws:s3:::bucktest1s/*\"]}]}"
   }
  ```

   it should return the Service Account credentials like:
   ```
   {
    "accessKey": "Y6ERKY9HLILEC9DMKDC5",
    "secretKey": "dYsoHwQKm+REFzNGkKyR4iLr5sSkZDJNSSMcdcIT"
   }
   ```
- Test that the Service Account can be added as a new host and have the proper permissions
